web/src/views/SubmissionPortal/Components/SubmissionList.vue

```diff
@@ -80,26 +84,79 @@ export default defineComponent({
       router?.push({ name: 'Submission Context', params: { id: item.id } });
     }
 
+    function waitForDialogUpdate(): Promise<void> {
+      return new Promise<void>((resolve) => {
+        const intervalId = setInterval(() => {
+          if (dialogUpdated.value === true) {
+            clearInterval(intervalId);
+            resolve();
+          }
+        }, 100);
+      });
+    }
+
     const submission = usePaginatedResults(ref([]), api.listRecords, ref([]), itemsPerPage);
     watch(options, () => {
       submission.setPage(options.value.page);
       const sortOrder = options.value.sortDesc[0] ? 'desc' : 'asc';
       submission.setSortOptions(options.value.sortBy[0], sortOrder);
     }, { deep: true });
 
+    function deleteDialogUpdate(confirmation: boolean) {
+      deleteConfirmation.value = confirmation;
+      dialogUpdated.value = true;
+    }
+
+    async function handleDeleteSubmission(item: api.MetadataSubmissionRecord) {
+      deleteDialog.value = true;
+      await waitForDialogUpdate();
+      if (deleteConfirmation.value) {
+        deleteSubmission(item.id);
+        submission.setPage(options.value.page);
+      }
+
+      deleteDialog.value = false;
+      deleteConfirmation.value = false;
+      dialogUpdated.value = false;
+      return item;
+    }
+
+    async function handleOverflowMenu(item: api.MetadataSubmissionRecord, title: String) {
+      switch (title) {
+        case 'Delete':
+          if (getStatus(item).text === 'in-progress') {
+            handleDeleteSubmission(item);
+          }
+          break;
+        default:
+          break;
+      }
+    }
+
     return {
       HARMONIZER_TEMPLATES,
+      deleteDialog,
+      deleteConfirmation,
+      dialogUpdated,
       IconBar,
       IntroBlurb,
       TitleBanner,
       createNewSubmission,
       getStatus,
       resume,
+      handleOverflowMenu,
+      deleteDialogUpdate,
       headers,
       options,
       submission,
     };
   },
+  data: () => ({
+    OverFlowMenuItems: [
+      { title: 'Delete' },
+      { title: 'Future button' },
+    ],
+  }),
```

Comment:
My initial thought with this structure was that it could make adding list options easier in the future and that's why it's set up to be kind of dynamic, but in hindsight I agree. I don't think it actually makes adding things any easier really since extra work would be required in the `handleOverflowMenu` code too.

Reply:
Added this change